framework/BaseClasses/BaseInterface.py

```diff
@@ -41,56 +43,99 @@ def __init__(self):
     self.metadataKeys     = set()                                                       # list of registered metadata keys to expect from this entity
     self.metadataParams   = {}                                                          # dictionary of registered metadata keys with repect to their indexes
 
-  def readXML(self, xmlNode, variableGroups=None, globalAttributes=None):
+  def handleInput(self, paramInput, variableGroups=None, globalAttributes=None):
     """
-      provide a basic reading capability from the xml input file for what is common to all types in the simulation than calls _readMoreXML
+      provide a basic reading capability from the xml input file for what is common to all types in the simulation than calls _handleInput
       that needs to be overloaded and used as API. Each type supported by the simulation should have: name (xml attribute), type (xml tag),
       verbosity (xml attribute)
-      @ In, xmlNode, ET.Element, input xml
+      @ In, paramInput, InputParameter, input data from xml
```

Comment:
please update the type for paramInput

Reply:
This is copied directly from the original BaseEntity. What would like to see it read instead?

Reply:
In hopes it covers the concern, I changed the type to `utils.InputData.parameterInput`.